File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
 twoSum(nums, target) {
  let indexed = nums.map((num, idx) => [num, idx]);
  indexed.sort((a, b) => a[0] - b[0]); // ordenar por valor

  let left = 0;
  let right = indexed.length - 1;

  while (left < right) {
    let sum = indexed[left][0] + indexed[right][0];
    if (sum === target) {
      return [indexed[left][1], indexed[right][1]];
    } else if (sum > target) {
      right--;
    } else {
      left++;
    }
  }
}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.Length - 1;
        int[] res = new int[2];

        while (l < r) {
            var sum = numbers[l] + numbers[r];

            if (sum == target) {
                res[0] = l + 1;
                res[1] = r + 1;
                break;
            } else if (sum > target) {
                r--;

            } else {
                l++;
            }
        }
    
        return res;
    
    }
}
